helpers/chatHelper.py

```diff
@@ -266,462 +264,629 @@ async def partChannel(
+        return ...
+
+    if recipient_token["tournament"]:
+        logger.warning(
+            "User attempted to send a message to a tournament client",
+            extra={
+                "token_id": user_token["token_id"],
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "recipient": send_to,
+            },
+        )
+        return ...
+
+    if await osuToken.isSilenced(recipient_token["token_id"]):
+        await osuToken.enqueue(
+            user_token["token_id"],
+            serverPackets.targetSilenced(recipient_token["username"]),
+        )
+
+        logger.warning(
+            "User tried to send a message to a silenced user",
+            extra={
+                "token_id": user_token["token_id"],
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "recipient_username": recipient_token["username"],
+                "recipient_user_id": recipient_token["user_id"],
+            },
+        )
+
+        return ...
+
+    if osuToken.is_restricted(recipient_token["privileges"]):
+        logger.warning(
+            "User attempted to send a message to a restricted user",
+            extra={
+                "token_id": user_token["token_id"],
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "recipient_username": recipient_token["username"],
+                "recipient_user_id": recipient_token["user_id"],
+            },
+        )
+        return ...
+
+    # TODO: check if user has us blacklisted
+    # fmt: off
+    if (
+        recipient_token["block_non_friends_dm"]
+        and user_token["user_id"]
+        not in await userUtils.getFriendList(recipient_token["user_id"])
+    ):
+    # fmt: on
+        await osuToken.enqueue(
+            user_token["token_id"],
+            serverPackets.targetBlockingDMs(recipient_token["username"]),
+        )
+
         logger.warning(
             "User tried to send a message to a user that is blocking non-friends dms",
             extra={
-                "username": userToken["username"],
-                "user_id": userToken["user_id"],
-                "recipient": to,
+                "token_id": user_token["token_id"],
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "recipient_username": recipient_token["username"],
+                "recipient_user_id": recipient_token["user_id"],
+            },
+        )
+
+        return ...
+
+    if await osuToken.awayCheck(
+        recipient_token["token_id"],
+        user_token["user_id"],
+    ):
+        await enqueue_private_message(
+            user_token=recipient_token,
+            recipient_token=user_token,
+            message=f"\x01ACTION is away: {recipient_token['away_message'] or ''}\x01",
```

Comment:
does awayCheck() require an away message? (is this None case possible at runtime?)

it will look a bit weird in the UI
`*cmyui is away: `

Reply:
Yeah it is:

https://github.com/osuAkatsuki/bancho-service/blob/63ebc92487e2c087f496495911ce31d8f704da78/objects/osuToken.py#L1188-L1189

I'll remove the `or ''` here